src/components/Wearables/WearableLifecycleProvider.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import { useCallback } from 'react';
+import {
+  NativeWearableLifecycleHandler,
+  WearableIntegration,
+  WearableIntegrationStatus,
+  EHRType,
+} from './WearableTypes';
+import { sortBy } from 'lodash';
+import {
+  PlatformLifecycleHandlersProvider,
+  platformLifecycleHandlers,
+} from './PlatformLifecycleHandlersProvider';
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const WearableLifecycleContext = React.createContext<{
+  onPreToggle: (
+    wearable: WearableIntegration,
+    enabled: boolean,
+  ) => Promise<void>;
+  onPostToggle: (wearable: WearableIntegration) => Promise<void>;
+  sanitizeEHRs: (
+    ehrs: WearableIntegration[],
+    enableMultiWearable?: boolean,
+    legacySort?: boolean,
+  ) => Promise<WearableIntegration[]>;
```

Comment:
These lifecycle methods were the only ones needed by HealthKit but we could add to these to support other native functionality or customers as needed.